lib/content/routes.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+/**
+ * @fileoverview Content system routing functions
+ *
+ */
+const config = require('config');
+const types = require('punchcard-content-types');
+const _ = require('lodash');
+
+const database = require('../database');
+const utils = require('../utils');
+
+/**
+ * All Content-Types Landing page
+ * @param {object} res - HTTP Response
+ * @param {object} allTypes all content types available to the CMS
+ */
+const all = (res, allTypes) => {
+  res.render('content/home', {
+    content: {
+      home: config.content.home,
+      base: config.content.base,
+      types: allTypes,
+    },
+  });
+};
+
+/**
+ * Individual Content-Types Landing page
+ * @param {object} res - HTTP Response
+ * @param {object} type - Content type configuration, after merged with input plugins
+ *
+ * @returns {promise} database promise returned
+ */
+const one = (res, type) => {
+  return database
+    .distinct(database.raw('ON (id) id'))
+    .select('*')
+    .from(`content-type--${type.id}`)
+    .orderBy('id', 'DESC')
+    .orderBy('revision', 'DESC').then(rws => {
```

Comment:
Check out this query Ayush and I came up with yesterday. It's used in enough places (this makes two) that we should probably abstract it out:

```SQL
select * from "content-type--services" where
revision in (select max("revision") from "content-type--services" group by "id") 
```
It selects all fields from content types where the revision is the highest revision, grouped by unique IDs. This will just return the most recent revisions of each piece of content.